optimalGuess.js

```diff
@@ -0,0 +1,111 @@
+//Comments will be in reference to the code in the line beneath it
+
+const readline = require('readline');
+
+const rl = readline.createInterface(process.stdin, process.stdout);
+
+function ask(questionText) {
+	return new Promise((resolve, reject) => {
+		rl.question(questionText, resolve);
+	});
+}
+
+// Function used to form initial guess ad adapt the range for optimal accuracy as guessing continues
+
+async function guessFcn(maxGuess, minGuess) {
+  //establishes range that will adapt as the guesses narrow the max and min. Will make the optimal guess in hte middle of the range.
+	const maxNumber = Number.parseFloat(maxGuess);
+
+	const minNumber = Number.parseFloat(minGuess);
+
+	const difference = maxNumber - minNumber;
+
+	//takes the range, applies a moving floor and guesses the value in the middle thus halfing the range each time.
+
+	const optimalGuess = Number.parseFloat(difference / 2 + (minNumber));
```

Comment:
Good, this works. If you use
```js
Math.floor(difference / 2 + minNumber)
```
you could skip assigning the extra variable to optimalGuess. Or you could use
```js
let guess = Math.floor((minGuess + maxGuess) / 2)
```
which would do the same thing